src/integrationTest/java/com/mongodb/hibernate/query/select/SortingSelectQueryIntegrationTests.java

```diff
@@ -0,0 +1,218 @@
+    }
+
+    @Test
+    void testOrderByMultipleFieldsWithoutTies() {
+        assertSelectionQuery(
+                "from Book where outOfStock = false ORDER BY title ASC, publishYear DESC, id ASC",
+                Book.class,
+                "{ 'aggregate': 'books', 'pipeline': [ {'$match': {'outOfStock': {'$eq': false}}}, { '$sort': { 'title': 1, 'publishYear': -1, '_id': 1 } }, {'$project': {'_id': true, 'discount': true, 'isbn13': true, 'outOfStock': true, 'price': true, 'publishYear': true, 'title': true} } ] }",
+                getBooksByIds(3, 2, 4, 5));
+    }
+
+    @Test
+    void testOrderByMultipleFieldsWithTies() {
+        assertSelectionQuery(
+                "from Book ORDER BY title ASC, publishYear DESC, id ASC",
+                Book.class,
+                "{ 'aggregate': 'books', 'pipeline': [ { '$sort': { 'title': 1, 'publishYear': -1, '_id': 1 } }, {'$project': {'_id': true, 'discount': true, 'isbn13': true, 'outOfStock': true, 'price': true, 'publishYear': true, 'title': true} } ] }",
+                resultList -> assertThat(resultList)
+                        .satisfiesAnyOf(
+                                list -> assertResultListEquals(getBooksByIds(3, 2, 4, 1, 5), list),
+                                list -> assertResultListEquals(getBooksByIds(3, 2, 4, 5, 1), list)));
+    }
+
+    @Test
+    void testSortFieldByAlias() {
+        assertSelectionQuery(
+                "select b.title as title, b.publishYear as year from Book as b ORDER BY year DESC, title ASC",
+                Object[].class,
+                "{'aggregate': 'books', 'pipeline': [{'$sort': {'publishYear': -1, 'title': 1}}, {'$project': {'title': true, 'publishYear': true}}]}",
+                List.of(
+                        new Object[] {"War and Peace", 2025},
+                        new Object[] {"The Brothers Karamazov", 1880},
+                        new Object[] {"Anna Karenina", 1877},
+                        new Object[] {"War and Peace", 1869},
+                        new Object[] {"Crime and Punishment", 1866}));
+    }
+
+    @Test
+    void testSortFieldByOrdinalReference() {
+        assertSelectionQuery(
+                "select b.title as title, b.publishYear as year from Book as b ORDER BY 1 ASC, 2 DESC",
+                Object[].class,
+                "{'aggregate': 'books', 'pipeline': [{'$sort': {'publishYear': -1, 'title': 1}}, {'$project': {'title': true, 'publishYear': true}}]}",
+                List.of(
+                        new Object[] {"Anna Karenina", 1877},
+                        new Object[] {"Crime and Punishment", 1866},
+                        new Object[] {"The Brothers Karamazov", 1880},
+                        new Object[] {"War and Peace", 2025},
+                        new Object[] {"War and Peace", 1869}));
+    }
+
+    @Nested
+    @DomainModel(annotatedClasses = Book.class)
+    @ServiceRegistry(settings = @Setting(name = DEFAULT_NULL_ORDERING, value = "first"))
```

Comment:
This is an approach I found we could limit Hibernate property configuration in one testing method (the setting won't be applied to the whole class for it would break all the testing cases) using `hibernate-testing` and JUnit5's `@Nested` features.